dot/parachain/overseer/mockable_overseer.go

```diff
@@ -50,14 +48,15 @@ func (m *MockableOverseer) Start() error {
 		sub.Run(m.ctx, overseerToSubSystem, m.SubsystemsToOverseer)
 	}(m.subSystem, m.overseerToSubsystem)
 
-	go m.processMessages()
+	// go m.processMessages()
```

Comment:
@kishansagathiya 

Couldn't figure out how to use this properly.
to mock the actions of the overseer(and other subsystems), I have goroutines in the test.

Reply:
This is how I have used it to write a test in collator protocol.

```
		        mockOverseer := overseer.NewMockableOverseer(t)
			overseerToSubsystem := mockOverseer.RegisterSubsystem(&cpvs)
			cpvs.OverseerToSubSystem = overseerToSubsystem
			cpvs.SubSystemToOverseer = mockOverseer.GetSubsystemToOverseerChannel()

			mockOverseer.Start()
			defer mockOverseer.Stop()

			mockOverseer.ExpectMessageWithAction(c.expectedMessage, func(msg any) {
				// do nothing
			})
			msg := collatorprotocolmessages.NewCollationProtocol()
			vdtChild := collatorprotocolmessages.NewCollatorProtocolMessage()
```

Let me look at your code and see if I can use mockable overseer there.

Reply:
I had to change the logic of the `processMessages()` method of mockable overseer.
I have pushed the updated code.